src/Components/Server/src/Builder/ServerComponentsEndpointOptions.cs

```diff
@@ -22,23 +22,35 @@ public class ServerComponentsEndpointOptions
     /// <para>
     /// A content security policy provides defense against security threats that can occur if
     /// the app uses compression and can be embedded in other origins. When compression is enabled,
-    /// embedding the app inside an <c>iframe</c> from other origins is prohibited.
+    /// embedding the app inside an <c>iframe</c> from other origins is forbidden.
     /// </para>
     /// <para>
     /// For more details see the security recommendations for Interactive Server Components in
     /// the official documentation.
     /// </para>
     /// </remarks>
-    public string? ContentSecurityFrameAncestorPolicy { get; set; } = "'self'";
+    public string? ContentSecurityFrameAncestorsPolicy { get; set; } = "'self'";
+
+    /// <summary>
+    /// Gets or sets a value that determines if the websocket compression should be disabled.
+    /// </summary>
+    /// <remarks>
+    /// Websocket compression is enabled by default, but it can be disabled by setting this value to <see langword="true" />.
+    /// When a callback for <see cref="ConfigureWebSocketAcceptContext"/> is provided, the value of this property will be
+    /// ignored, whether compression is enabled or not will be determined by the callback, and the Content Security Policy
+    /// will be applied according to the value of <see cref="ContentSecurityFrameAncestorsPolicy"/>.
+    /// When compression is disabled and no callback is provided, the Content Security Policy header will not be set on the
+    /// responses.
+    /// </remarks>
+    public bool DisableWebSocketCompression { get; set; }
 
     /// <summary>
     /// Gets or sets a function to configure the <see cref="WebSocketAcceptContext"/> for the websocket connections
     /// used by the server components.
-    /// By default, a policy that enables compression and sets a Content Security Policy for the frame ancestors
-    /// defined in <see cref="ContentSecurityFrameAncestorPolicy"/> will be applied.
     /// </summary>
-    public Func<HttpContext, WebSocketAcceptContext>? ConfigureWebsocketOptions { get; set; } = EnableCompressionDefaults;
-
-    private static WebSocketAcceptContext EnableCompressionDefaults(HttpContext context) =>
-        new() { DangerousEnableCompression = true };
+    /// <remarks>
+    /// By default, a policy that enables compression and sets a Content Security Policy for the frame ancestors
+    /// defined in <see cref="ContentSecurityFrameAncestorsPolicy"/> will be applied.
+    /// </remarks>
+    public Func<HttpContext, WebSocketAcceptContext, Task>? ConfigureWebSocketAcceptContext { get; set; }
```

Comment:
I think the approved API made this non-nullable and gave it a default value of `(httpContext, webSocketAcceptContext) => Task.CompletedTask`. It's probably fine to make this null by default, but just want to call it out in case this was a mistake.

Reply:
Ohh, is the reason why this was made `null` because then you can easily detect the case when you definitely _don't_ need to add the CSP header? That makes sense to me if so.

Reply:
Yes